chromadb/test/property/test_add.py

```diff
@@ -86,3 +86,23 @@ def test_out_of_order_ids(api: API) -> None:
     coll.add(ids=ooo_ids, embeddings=embeddings)
     get_ids = coll.get(ids=ooo_ids)["ids"]
     assert get_ids == ooo_ids
+
+
+def test_add_partial(api: API) -> None:
+    """Tests adding a record set with some of the fields set to None."""
+
+    api.reset()
+
+    coll = api.create_collection("test")
+    # TODO: We need to clean up the api types to support this typing
+    coll.add(
+        ids=["1", "2", "3"],
+        embeddings=[[1, 2, 3], [1, 2, 3], [1, 2, 3]],
+        metadatas=[{"a": 1}, None, {"a": 3}],  # type: ignore
+        documents=["a", "b", None],  # type: ignore
```

Comment:
whats happening here? will users have the type checker yell at them?

Reply:
It would have always yelled at them :( Our types don't support this partial application (and metadata type is broked) but we logically have supported it and thats what people were relying on that broke. It makes sense to support this, many cases where maybe one data point does not contain metadata.

We need to go over our types and fix them.